### PR TITLE
fix: update paragon and country field

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description: Update Paragon version and changed the country field implementation as per documentation VAN-1814

[JIRA](https://2u-internal.atlassian.net/browse/VAN-1814)

How Has This Been Tested?

It has been tested locally

|After|
|-----|
  
image

### CODE REVIEW

File: src/register/RegistrationFields/CountryField/CountryField.jsx [hunk 1]
```diff
@@ -31,6 +31,12 @@ const CountryField = (props) => {
   } = props;
   const { formatMessage } = useIntl();
   const dispatch = useDispatch();
+
+  const countryFieldValue = {
+    userProvidedText: selectedCountry.displayValue,
+    selectionValue: selectedCountry.countryCode,
+    selectionId: selectedCountry.countryCode,
+  };
```
Comment (syedsajjadkazmii) [inline src/register/RegistrationFields/CountryField/CountryField.jsx:39]: [NIT] Can we add extra/empty line after this?

File: src/register/RegistrationFields/CountryField/CountryField.jsx [hunk 2]
```diff
@@ -49,7 +55,7 @@ const CountryField = (props) => {
         { target: { name: 'country' } },
         { countryCode, displayValue: countryDisplayValue },
       );
-    } else if (!selectedCountry.countryCode) {
+    } else if (!selectedCountry.displayValue) {
```
Comment (syedsajjadkazmii) [inline src/register/RegistrationFields/CountryField/CountryField.jsx:58]: why did we change this?

File: src/register/RegistrationFields/CountryField/CountryField.jsx [hunk 3]
```diff
@@ -65,30 +71,27 @@ const CountryField = (props) => {
...
-
   const handleOnFocus = (event) => {
     handleErrorChange('country', '');
     dispatch(clearRegistrationBackendError('country'));
     onFocusHandler(event);
   };
 
   const handleOnChange = (value) => {
-    onChangeHandler({ target: { name: 'country' } }, { countryCode: '', displayValue: value });
+    onChangeHandler({ target: { name: 'country' } }, { countryCode: value.selectionId, displayValue: value.userProvidedText });
```
Comment (syedsajjadkazmii) [inline src/register/RegistrationFields/CountryField/CountryField.jsx:87]:
```suggestion
    onChangeHandler({ target: { name: 'country' } }, { countryCode: value.selectionValue, displayValue: value.userProvidedText });
```

File: src/register/RegistrationFields/CountryField/CountryField.jsx [hunk 4]
```diff
@@ -65,30 +71,27 @@ const CountryField = (props) => {
...
+    if (value.selectionId !== '') {
+      handleOnBlur({ target: { name: 'country', value: value.userProvidedText } });
+    }
```
Comment (syedsajjadkazmii) [inline src/register/RegistrationFields/CountryField/CountryField.jsx:90]: What scenario is being handled here?

File: src/register/RegistrationFields/CountryField/CountryField.jsx [hunk 5]
```diff
@@ -65,30 +71,27 @@ const CountryField = (props) => {
 
     const { value } = event.target;
 
-    const { countryCode, displayValue, error } = validateCountryField(
+    const { error } = validateCountryField(
       value.trim(), countryList, formatMessage(messages['empty.country.field.error']), formatMessage(messages['invalid.country.field.error']),
     );
-
-    onChangeHandler({ target: { name: 'country' } }, { countryCode, displayValue });
```
Comment (syedsajjadkazmii) [inline src/register/RegistrationFields/CountryField/CountryField.jsx:72]: why is this removed?

**Review by syedsajjadkazmii — COMMENTED** [review]
(After approval) Can we coordinate with Frontend-BOM regarding this paragon update for additional safeguard?

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 4 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:90]: If we click the dropdown icon and select the item from list that mean we are not focusing the input box we just use the dropdown icon and select the element

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 5 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:72]: Now we are calling this onChangeHandler inside the onChange function and it does not make sense to keep it here inside the blur event

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 3 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:87]: This selectionValue does not have country code it keep display value which is normal country name not its code that is why I have to use sectionId instead of selectionValue

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 2 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:58]: if we put some wrong value inside the country field and we move around the login and signup tabs then this !selectedCountry.countryCode becomes always true and onChangeHandler set the country field with empty data as we are calling this onChangeHandler with blank data so this reset the country field, actual behaviour is to keep whatever user have input in country field should be there in tabs switch

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 3 above)
Comment (syedsajjadkazmii) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:87]: then for consistency we should also change this (set displayValue in selectionValue)

```

  const countryFieldValue = {
    userProvidedText: selectedCountry.displayValue,
    selectionValue: selectedCountry.displayValue,
    selectionId: selectedCountry.countryCode,
  };
```

File: src/register/RegistrationFields/CountryField/CountryField.jsx (hunk 4 above)
Comment (syedsajjadkazmii) [reply src/register/RegistrationFields/CountryField/CountryField.jsx:90]: isn't the `selectionId` provided on item click? i wanted to understand this check expecially.

**Review by syedsajjadkazmii — COMMENTED** [review]
Some thoughts/questions. Thankyou

**Review by syedsajjadkazmii — APPROVED** [review]
